### PR TITLE
feat: JSON Symbol library and symbol node styling

### PR DESCRIPTION
**Pull request by lorentzfb**
Initially, this PR was about improving styling of connector symbol nodes in goGraph. During this work, progress was made on the spec for the "smart" connector symbol SVG, and new symbols on this format was available to test. This led to more work especially on the symbol api implementation.

### Changes 

- New symbol library setup: Use a single json file as the symbol source (symbol-library.json). This file can be generated by using the [Bo.Cli ](https://github.com/equinor/dugtrio-experimental/tree/main/bo) which batch processes SVGs on the new format and spits out a json file with a dictionary structure of symbols.  Previously, we had one .ts file for each symbol. The new solution  is more future proof in the sense that json is more likely to be the symbol exchange format.
- Old symbol library system removed.
- Added a storybook story for the symbol api to show all available symbols using `RdfGoGraph` (StoryWrapper for GoGraph reused). In this story, all symbols in symbol-library.json are converted to turtle at runtime which at the end becomes quads that are dispatched as a RdfActions. This gives us a useful overview of all symbols, and do visual/functional tests of new symbols with minimal effort.
- Symbol node template updated to handle new symbol format
- Added color and hover effects to the symbol node template
- CyGraph updated to handle new symbol api. 90% success rate (see further work below)
- Tests in core updated to handle new symbol api

### Further work

- Connector Symbol rotation does not work properly in CyGraph. The issue was introduced by the fact that the new symbol API can return symbols where width != height (we only used square shaped symbols in Cy and did not resolve this issue). Currently, width and height does not change when the symbol is rotated, and the symbol is clipped. The easiest fix is to allow only angles of `n x 90` and then just flip width and height on rotation:

```js
// Original w and h (at 0deg rotation)
const w = 200;
const h = 400;

// Width and height adjusted for 90deg rotation
const angle = 90;
const flip = (angle / 90) % 2 === 1;
const w_adj = flip ? h : w; // 400
const h_adj = flip ? w : h; // 200
```

### CODE REVIEW

File: src/cyGraph/storybook/CyGraph.stories.tsx [hunk 1]
```diff
@@ -31,32 +31,38 @@ const storyTurtle = `
...
         ui:hasConnector  example:Separatox3001LiquidOutput , example:Separatox3001GasOutput ;
-        ui:hasSvg              "Separator_1" .
+        ui:hasSvg              "PT003A" .
+
+example:Separatox3001LiquidOutput2
+        ui:hasConnectorSuffix  "4" .
 
 example:Separatox3001LiquidOutput
-        ui:hasConnectorSuffix  "c3" .
+        ui:hasConnectorSuffix  "3" .
```
Comment (toftalisk) [inline src/cyGraph/storybook/CyGraph.stories.tsx:59]: Why was this changed?

File: src/cyGraph/storybook/StoryWrapper.tsx [hunk 2]
```diff
@@ -19,6 +21,18 @@ type Property = {
...
 
+export function cySymbolProvider(id: string, rotation?: number): UiNodeSymbol | undefined {
+	const symbol = getConnectorSymbolAdvanced(id, {
+		rotation: rotation,
+		mutateConnectorRelativePosition: 'CenterCenter',
+		mutateSvgStringOnRotation: true,
+		mutateConnectorRelativePositionOnRotation: true,
+	});
+	if (!symbol) return;
+	console.log(ConnectorSymbolToUiNodeSymbol(symbol));
```
Comment (toftalisk) [inline src/cyGraph/storybook/StoryWrapper.tsx:32]:
```suggestion
```

File: src/goGraph/node-templates/symbol-node-template.ts [hunk 3]
```diff
@@ -1,19 +1,121 @@
-import go from 'gojs';
+import go, { GraphObject } from 'gojs';
 import { portDirectionToSpot } from '../item-templates/position-port-item-template';
 
+// this is shown by the mouseHover event handler
+var nodeHoverMenu = new go.Adornment('Spot', {
```
Comment (toftalisk) [inline src/goGraph/node-templates/symbol-node-template.ts:5]: This file is getting big. Can we separate nodeHoverMenu into separate file, with an exporting function returning it?

File: src/goGraph/node-templates/symbol-node-template.ts [hunk 4]
```diff
@@ -28,25 +130,95 @@ export function createSymbolNodeTemplate(clickHandler?: ((e: go.InputEvent, this
...
+					background: colorizeLayers ? 'orange' : 'transparent',
+					alignment: go.Spot.Center,
+				})
+					.bind('width', 'symbolWidth', (w) => w + nodePadding * 2)
+					.bind('height', 'symbolHeight', (h) => h + nodePadding * 2)
+			)
+
+			// TOP LABEL
+			.add(
+				new go.Panel(go.Panel.Auto, {
```
Comment (toftalisk) [inline src/goGraph/node-templates/symbol-node-template.ts:145]: Can this panel go into separate file?

**Review by toftalisk — COMMENTED** [review]
This is starting to look really good 👍 

Had some ideas for further refactoring as one of the files got a bit big. No need to fix no.

Good idea with cytoscape height / width flipping. I will see if  I can fix it later.

Awesome 🎉

**Review by toftalisk — APPROVED** [review]
Awesome 🎉

File: src/cyGraph/storybook/CyGraph.stories.tsx (hunk 1 above)
Comment (lorentzfb) [reply src/cyGraph/storybook/CyGraph.stories.tsx:59]: Because the connector ids in the new svg files are integers. Could add "c" prefix at runtime though.

File: src/cyGraph/storybook/StoryWrapper.tsx (hunk 2 above)
Comment (lorentzfb) [reply src/cyGraph/storybook/StoryWrapper.tsx]: [eff82b3](https://github.com/equinor/rdf-graph/pull/229/commits/eff82b307927896c71ea434bbb61052ede7a4271)

File: src/goGraph/node-templates/symbol-node-template.ts (hunk 3 above)
Comment (lorentzfb) [reply src/goGraph/node-templates/symbol-node-template.ts:5]: I will tidy up this file in a later PR

File: src/goGraph/node-templates/symbol-node-template.ts (hunk 4 above)
Comment (lorentzfb) [reply src/goGraph/node-templates/symbol-node-template.ts:145]: Yes. I will tidy up this file in a later PR.

File: src/goGraph/node-templates/symbol-node-template.ts (hunk 3 above)
Comment (toftalisk) [reply src/goGraph/node-templates/symbol-node-template.ts:5]: 👍

File: src/goGraph/node-templates/symbol-node-template.ts (hunk 4 above)
Comment (toftalisk) [reply src/goGraph/node-templates/symbol-node-template.ts:145]: 👍
